optlang/glpk_interface.py

```diff
@@ -579,6 +579,9 @@ def __setstate__(self, repr_dict):
         if repr_dict['glpk_status'] == 'optimal':
             self.optimize()  # since the start is an optimal solution, nothing will happen here
 
```

Comment:
Ah ok, good catch. Maybe wrap glp_prob into its own class with a `__del__` method, use the wrapped object as Model.problem and rather reference Model instead of Model.problem in the objective?